pkg/internal/apis/config/validate.go

```diff
@@ -155,6 +155,11 @@ func validatePortMappings(portMappings []PortMapping) error {
 		addr := net.ParseIP(portMapping.ListenAddress)
 		addrString := addr.String()
 
+		if (portMapping.HostPort == -1 || portMapping.HostPort == 0) && (portMapping.Protocol == PortMappingProtocolTCP || portMapping.Protocol == "") {
+			// Port -1 and 0 cause a random port to be selected, thus duplicates are allowed
+			continue
+		}
```

Comment:
this does not look great to me, seems very hacky

Reply:
Do you have a suggestion on how to improve this condition? The goal is to match TCP port 0 and -1 which already were considered special cases by Kind. 

I noticed that the unit tests set the portMapping Protocol to an empty string, and when parsing my config file it's "TCP", which is what this condition checks for.

Reply:
the problem is the feature itself is very hacky https://github.com/kubernetes-sigs/kind/issues/3512#issuecomment-1936124620 , you are delegating something to kind "assign random ports to container ports" for later require an external tool to return those values

Reply:
This feature already existed before this PR. I did not introduce the feature. I am merely allowing it to be used multiple times.

Reply:
yeah, we talked on the referenced issue, I'm going to resolve this for the sake of starting the conversation from scratch